setImmediate.js

```diff
@@ -35,13 +32,13 @@
         if (currentlyRunningATask) {
             // Delay by doing a setTimeout. setImmediate was tried instead, but in Firefox 7 it generated a
             // "too much recursion" error.
-            setTimeout(partiallyApplied(runIfPresent, handle), 0);
+            setTimeout(function () { runIfPresent(handle); }, 0);
         } else {
             var task = tasksByHandle[handle];
             if (task) {
                 currentlyRunningATask = true;
                 try {
-                    task();
+                    run(task);
```

Comment:
Separate function, outside of `try finally` [for performance](https://github.com/petkaantonov/bluebird/wiki/Optimization-killers#2-unsupported-syntax).